BUG - hooks not working on windows 10, when user account name contains non-ascii characters

When user account name contains non-ascii characters such as 'š', such that python executable ends up for example in C:\Users\john.š\\.cache\pre-commit\repo\py_env-python3.8\Scripts\python.exe, when committing to the git repository following message appears:

An unexpected error has occurred: AssertionError: BUG: expected environment for python to be healthy() immediately after install, please open an issue describing your environment.

PS: fucntion os.path.isfile() in parse_shebang.normexe() returns False, even though the executable exists there and is a file.

### version information

```
pre-commit version: 2.9.0
sys.version:
    3.8.0 (tags/v3.8.0:fa919fd, Oct 14 2019, 19:37:50) [MSC v.1916 64 bit (AMD64)]
sys.executable: c:\users\juraj.š\projects\aiscr-amcr-webclient\venv\scripts\python.exe
os.name: nt
sys.platform: win32
```

### error information

```
An unexpected error has occurred: AssertionError: BUG: expected environment for python to be healthy() immediately after install, please open an issue describing your environment
```

```
Traceback (most recent call last):
  File "c:\users\juraj.š\projects\aiscr-amcr-webclient\venv\lib\site-packages\pre_commit\error_handler.py", line 65, in error_handler
    yield
  File "c:\users\juraj.š\projects\aiscr-amcr-webclient\venv\lib\site-packages\pre_commit\main.py", line 347, in main
    return hook_impl(
  File "c:\users\juraj.š\projects\aiscr-amcr-webclient\venv\lib\site-packages\pre_commit\commands\hook_impl.py", line 217, in hook_impl
    return retv | run(config, store, ns)
  File "c:\users\juraj.š\projects\aiscr-amcr-webclient\venv\lib\site-packages\pre_commit\commands\run.py", line 401, in run
    install_hook_envs(hooks, store)
  File "c:\users\juraj.š\projects\aiscr-amcr-webclient\venv\lib\site-packages\pre_commit\repository.py", line 213, in install_hook_envs
    _hook_install(hook)
  File "c:\users\juraj.š\projects\aiscr-amcr-webclient\venv\lib\site-packages\pre_commit\repository.py", line 88, in _hook_install
    raise AssertionError(
AssertionError: BUG: expected environment for python to be healthy() immediately after install, please open an issue describing your environment
```

--- a/pre_commit/languages/python.py
+++ b/pre_commit/languages/python.py
@@ -36,7 +36,7 @@ def _version_info(exe: str) -> str:
 
 def _read_pyvenv_cfg(filename: str) -> Dict[str, str]:
     ret = {}
-    with open(filename) as f:
+    with open(filename, encoding='UTF-8') as f:
         for line in f:
             try:
                 k, v = line.split('=')

--- a/tests/languages/python_test.py
+++ b/tests/languages/python_test.py
@@ -23,6 +23,13 @@ def test_read_pyvenv_cfg(tmpdir):
     assert python._read_pyvenv_cfg(pyvenv_cfg) == expected
 
 
+def test_read_pyvenv_cfg_non_utf8(tmpdir):
+    pyvenv_cfg = tmpdir.join('pyvenv_cfg')
+    pyvenv_cfg.write_binary('hello = hello john.š\n'.encode())
+    expected = {'hello': 'hello john.š'}
+    assert python._read_pyvenv_cfg(pyvenv_cfg) == expected
+
+
 def test_norm_version_expanduser():
     home = os.path.expanduser('~')
     if os.name == 'nt':  # pragma: nt cover

